src/exoticatechnologies/hullmods/exotics/HullmodExoticHandler.kt

```diff
@@ -0,0 +1,169 @@
+package exoticatechnologies.hullmods.exotics
+
+import com.fs.starfarer.api.combat.ShipVariantAPI
+import com.fs.starfarer.api.fleet.FleetMemberAPI
+import exoticatechnologies.modifications.exotics.impl.HullmodExotic
+import exoticatechnologies.util.AnonymousLogger
+import exoticatechnologies.util.datastructures.Optional
+
+object HullmodExoticHandler {
+    private val lookupMap: HashMap<HullmodExoticKey, HullmodExoticInstallData> = hashMapOf()
+
+    /**
+     * Method for checking whether the [HullmodExotic] should be installed onto a [ShipVariantAPI] by checking
+     * if we hadn't already installed it or not
+     *
+     * @param hullmodExotic the HullmodExotic in question
+     * @param parentFleetMember the [FleetMemberAPI] of the root module, or the whole ship's if it's a single-module ship
+     * @param variant the variant to check
+     * @param variantList list of module variants belonging to the parent module (optional)
+     */
+    fun shouldInstallHullmodExoticToVariant(hullmodExotic: HullmodExotic, parentFleetMember: FleetMemberAPI, variant: ShipVariantAPI, variantList: Optional<List<ShipVariantAPI>>): Boolean {
+        AnonymousLogger.log("--> shouldInstallHullmodExoticToVariant()\thullmodExotic: ${hullmodExotic}, variant: ${variant}", "HullmodExoticHandler")
+        // First things first, check if we have this in the map
+        // we will do this by forming up a [HullmodExotica, FleetMemberAPI] pair as a key
+        val hullmodExoticKey = HullmodExoticKey(
+                hullmodExotic = hullmodExotic,
+                parentFleetMember = parentFleetMember
+        )
+        val currentInstallData = lookupMap[hullmodExoticKey]
+
+        // now, if we already have some install data, we are going to be checking the variant we're about to install
+        // versus two different things:
+        // 1. is it in the expected list?
+        // 2. is it already NOT in the installed list?
+        // we'll just do nothing if it fails either of these and return false.
+        val retVal = if (currentInstallData != null) {
+            AnonymousLogger.log("shouldInstallHullmodExoticToVariant()\talready had install data !!!", "HullmodExoticHandler")
+            // now, check if this variant is in the expected list
+            val expectedVariants = currentInstallData.listOfExpectedVariants
+//            val isExpected: Boolean = expectedVariants.contains(variant)
```

Comment:
delete

Reply:
done. resolving. 

all comments relating to `HullmodExoticHandler` are **most likely** either outdated or done.